src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.Analyzer.cs

```diff
@@ -129,6 +130,72 @@ protected override bool ReadOnlyFieldAllowed()
                 var scope = this.SelectionResult.GetContainingScopeOf<ConstructorDeclarationSyntax>();
                 return scope == null;
             }
+
+            protected override ITypeSymbol GetSymbolType(SemanticModel semanticModel, ISymbol symbol)
+            {
+                var selectionOperation = semanticModel.GetOperation(this.SelectionResult.GetContainingScope());
```

Comment:
You may want to add a comment as to what this whole thing is trying to achieve...

Reply:
tHe CoDe Is ThE dOcUmEnTaTiOn